packages/file-storage/test/file-storage-fs.e2e-spec.ts

```diff
@@ -0,0 +1,150 @@
+      expect(res.length).toBe(0);
+    });
+
+    it('uploadFile uploads a file', async () => {
+      await fileStorage.uploadFile({ filePath: testFileName, content: 'this is a test' });
+      const result = await fileStorage.readDir({ dirPath });
+      expect(result.length).toBe(1);
+      expect(result[0]).toBe(testFileName);
+    });
+
+    it('calling fileExists on a filepath that exists returns true', async () => {
+      const fileExists = await fileStorage.fileExists({ filePath: testFileName });
+      expect(fileExists).toBe(true);
+    });
+
+    it('deleteFile deletes a file', async () => {
+      await fileStorage.deleteFile({ filePath: testFileName });
+      const result = await fileStorage.readDir({ dirPath });
+      expect(result.length).toBe(0);
+    });
+
+    it('uploadStream uploads a file', async () => {
+      const upload = await fileStorage.uploadStream({ filePath: testFileName });
+      const entry = Readable.from(iterable);
+      await pipeline(entry, upload).catch((err) => {
+        console.error(err);
+      });
+      // add delay, otherwise test is flaky
+      await new Promise<void>((resolve) =>
+        setTimeout(async () => {
+          const result = await fileStorage.readDir({ dirPath });
+          expect(result.length).toBe(1);
+          resolve();
+        }, 100),
+      );
+    });
+
+    it('downloadFile downloads a file', async () => {
+      const file = await fileStorage.downloadFile({ filePath: testFileName });
+      expect(file.toString()).toBe(iterable.join(''));
+    });
+
+    it('downloadStream downloads a file', async () => {
+      const download = await fileStorage.downloadStream({ filePath: testFileName });
+      expect(download).toBeInstanceOf(Readable);
+      download.on('data', (chunk) => {
+        expect(chunk.toString()).toBe(iterable.join(''));
+      });
+    });
+
+    it('uploads a file to a nested folder', async () => {
+      if (storageType === StorageType.FS) await mkdir(nestedFilePath, { recursive: true });
+
+      await fileStorage.uploadFile({ filePath: `${nestedDir}/${nestedFileName}`, content: 'this is a nested file' });
+      const result = await fileStorage.readDir({ dirPath: nestedDir });
+      expect(result.length).toBe(1);
+      expect(result[0]).toBe(nestedFileName);
+    });
+
+    it('readDir returns an array of files and folders in a dir', async () => {
+      const result = await fileStorage.readDir({ dirPath });
+      expect(result.length).toBe(2);
+      expect(result).toEqual([nestedDir, testFileName]);
+    });
+
+    it('calling fileExists on a filepath that doesnt exist throws an error', async () => {
+      await expect(fileStorage.fileExists({ filePath: 'fileDoesntExist' })).rejects.toThrow();
+    });
+
+    it('deleteDir deletes a dir', async () => {
+      await fileStorage.deleteDir({ dirPath });
+      expect(await fileStorage.readDir({ dirPath })).toEqual([]);
+    });
+
+    afterAll(async () => {
```

Comment:
Also the S3 bucket cleanup should be done `afterAll`.
What's the reasoning with putting `afterAll` at the bottom ?